File: langs/java/TypeUnion.java
package ru.gbooking.apiv2;

import java.util.*;
import java.io.IOException;
import com.fasterxml.jackson.core.*;
import com.fasterxml.jackson.core.type.*;
import com.fasterxml.jackson.databind.*;
import com.fasterxml.jackson.databind.annotation.*;

@JsonDeserialize(using = TypeUnion.Deserializer.class)
@JsonSerialize(using = TypeUnion.Serializer.class)
public class TypeUnion {
    public List<TypeElement> enumArrayValue;
    public TypeElement enumValue;

    static class Deserializer extends JsonDeserializer<TypeUnion> {
        @Override
        public TypeUnion deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException, JsonProcessingException {
            TypeUnion value = new TypeUnion();
            switch (jsonParser.getCurrentToken()) {
            case START_ARRAY:
                value.enumArrayValue = jsonParser.readValueAs(new TypeReference<List<TypeElement>>() {});
                break;
            case VALUE_STRING:
                value.enumValue = jsonParser.readValueAs(TypeElement.class);
                break;
            default: throw new IOException("Cannot deserialize TypeUnion");
            }
            return value;
        }
    }

    static class Serializer extends JsonSerializer<TypeUnion> {
        @Override
        public void serialize(TypeUnion obj, JsonGenerator jsonGenerator, SerializerProvider serializerProvider) throws IOException {
            if (obj.enumArrayValue != null) {
                jsonGenerator.writeObject(obj.enumArrayValue);
                return;
            }
            if (obj.enumValue != null) {
                jsonGenerator.writeObject(obj.enumValue);
                return;
            }
            throw new IOException("TypeUnion must not be null");
        }
    }
}

File: langs/golang/typeunion.go
package ru.gbooking.apiv2

type TypeUnion struct {
	Enum      *TypeElement
	EnumArray []TypeElement
}

func (x *TypeUnion) UnmarshalJSON(data []byte) error {
	x.EnumArray = nil
	x.Enum = nil
	object, err := unmarshalUnion(data, nil, nil, nil, nil, true, &x.EnumArray, false, nil, false, nil, true, &x.Enum, false)
	if err != nil {
		return err
	}
	if object {
	}
	return nil
}

func (x *TypeUnion) MarshalJSON() ([]byte, error) {
	return marshalUnion(nil, nil, nil, nil, x.EnumArray != nil, x.EnumArray, false, nil, false, nil, x.Enum != nil, x.Enum, false)
}
